File: services/identity-service/src/main/java/br/com/deltaglobalbank/identity/features/signingKeys/rotateSigningKeys/SigningKeyView.java
package br.com.deltaglobalbank.identity.features.signingKeys.rotateSigningKeys;

import java.time.Instant;
import java.util.UUID;

public record SigningKeyView(
    UUID id,
    String kid,
    String algorithm,
    String status,
    Instant activatedAt,
    Instant retiredAt
) {
}

File: services/identity-service/src/main/kotlin/br/com/deltaglobalbank/identity/features/signingKeys/SigningKeyView.kt
package br.com.deltaglobalbank.identity.features.signingKeys

import br.com.deltaglobalbank.identity.domain.token.SigningKey
import br.com.deltaglobalbank.identity.features.signingKeys.rotateSigningKeys.SigningKeyView

fun SigningKey.toView() = SigningKeyView(
    id = id, kid = kid, algorithm = algorithm,
    status = status().toDatabaseValue(),
    activatedAt = activatedAt,
    retiredAt = snapshot().retiredAt,
)